orttraining/orttraining/core/graph/gradient_builder_base.h

```diff
@@ -169,11 +169,22 @@ class GradientBuilderBase {
     return ConstantValueNode(1.0f, "OneConstant");
   }
 
+  void ComputeBroadcastBackwardAxesDynamic(const ArgDef& a,
+                                           const ArgDef& b,
+                                           const ArgDef& a_axes,
```

Comment:
>const ArgDef& [](start = 43, length = 13)

I think we should change this to 
const ArgDef*
if the axes is not need, we can pass in nullptr;